src/handler/on-start/commands.handler.ts

```diff
@@ -0,0 +1,42 @@
+import { handlerFileFilter, handlerRoot } from "./constant";
+import { Client } from "discord.js";
+import { SlashCommand } from "../../types/utils";
+import { importFiles } from "../../utils/files-import";
+
+const loadSlashCommands = async (client: Client) => {
+  const commands = await importFiles<SlashCommand>({
+    path: `./${handlerRoot}/commands`,
+    options: {
+      fileFilter: [handlerFileFilter, "!*.type.ts", "!constant.ts"],
+    },
+  });
+  commands.forEach(({ data }) => {
+    if (!data.name) return;
+    client.slashCommands.set(data.name, data);
+  });
+
+  const testingCommands = commands.filter(({ data }) => !data.production);
+  const productionCommands = commands.filter(({ data }) => data.production);
+
+  // This is not recommended to set slash command on every restart
+  if (process.env.TESTING_GUILD_ID) {
+    await client.application?.commands.set(
+      testingCommands.map(({ data }) => ({
+        name: data.name,
+        description: data.description,
+      })),
+      process.env.TESTING_GUILD_ID,
+    );
+  }
+
+  await client.application?.commands.set(
+    productionCommands.map(({ data }) => ({
+      name: data.name,
+      description: data.description,
+    })),
+  );
+};
+
```

Comment:
After simply test, looks like this will run before bot login.

Reply:
You can take advantage of the well-crafted type within `discord.js`. If the type of `client` is `Client<true>`, it signifies that the `client` is ready, eliminating the need to use `?.` in `client.application?.commands`.

Reply:
what do you think if there is a prefix command to manual register the slash command if needed, because there is a daily limit of how much we can update slash command, so client will always be `Client<true>`

Reply:
it will looks something like this
<img width="487" alt="CleanShot 2023-08-14 at 22 25 22@2x" src="https://github.com/yyueniao/miters/assets/46621156/0d309d9d-22bf-4a02-8861-9234d0a75267">

Reply:
And could you explain more about your comment here?

Reply:
> what do you think if there is a prefix command to manual register the slash command if needed, because there is a daily limit of how much we can update slash command, so client will always be `Client<true>`

I am new to discord bot development, is (this)[https://stackoverflow.com/questions/75317190/discordapierror-maximum-number-of-application-commands-reached-100] what you mentioned?

Reply:
> so that we can have more control on which commands should be available globally or testing guild only without having to change the code

This sounds making sense to me, I will happy to see!

Reply:
> > so that we can have more control on which commands should be available globally or testing guild only without having to change the code
> 
> This sounds making sense to me, I will be happy to see!

Hello @yyueniao!  I have done the commands I mentioned above, you may have a look.

Reply:
Thanks @pwkang ! Such a huge update, I will need time to go through.

Reply:
I haven't gone through every files, but after tested, it looks intuitive. Good job!